rj_gameplay/stp/tactic/pass_or_shoot.py

```diff
@@ -0,0 +1,48 @@
+from typing import Optional
+
+import stp.tactic as tactic
+import stp.skill.ball_carrier as ball_carrier
+import stp.skill.seeker as seeker
+import stp.skill.capture as capture
+
+
+class Skills(tactic.SkillsEnum):
+    BALL_CARRIER = tactic.SkillEntry(ball_carrier.IBallCarrier)
+    SEEKERS = tactic.SkillEntry(seeker.ISeeker)
+    RECEIVER = tactic.SkillEntry(capture.ICapture)
+
+
+class PassOrShoot(tactic.ITactic):
+    """ Tactic that controls one ball carrier and multiple seekers.
+    """
+
+    __slots__ = ["skills", "BALL_CARRIER", "RECEIVER", "SEEKERS"]
+
+    def __init__(self, ctx: tactic.Ctx):
+        self.skills = Skills(ctx.skill_factory)
+
+        self.BALL_CARRIER = self.skills.BALL_CARRIER
+        self.RECEIVER = self.skills.RECEIVER
+        self.SEEKERS = self.skills.SEEKERS
+
+    def get_requests(self, prev_skills: tactic.SkillsDict) -> tactic.RoleRequests:
+        role_requests: tactic.RoleRequests = tactic.RoleRequests()
+
+        maybe_pass = self.get_pass(prev_skills)
+        if maybe_pass:
+            role_requests[self.RECEIVER] = self.RECEIVER.skill.create_request()
+
+        role_requests[self.BALL_CARRIER] = self.BALL_CARRIER.skill.create_request()
+        role_requests[self.SEEKERS] = self.SEEKERS.skill.create_request()
```

Comment:
How do we support multiple (an arbitrary number of) seekers in this paradigm?

Reply:
This is still WIP but I was thinking that the `RoleRequest` has a way of specifying the min / max number of robots it is accepting or something. So the `RoleRequest` that `self.SEEKERS.skill.create_request` returns will allow for accepting multiple robots.

Reply:
Why not just create multiple role requests?

Reply:
That could work as well. We would need to know how many robots there are though, so I guess we can pass world_state in here.

Reply:
That could work as well. We would need to know how many robots there are though, so I guess we can pass world_state in here.

Reply:
I think the original plan was that you just request as many roles as you can possibly use (but at lower priority), and then check if they're filled in the `tick` function, and then the roles are just stored as member variables. Is there a reason you moved away from this? (I don't love it given that it introduces additional statefullness, so if you have better ideas I'd love to hear them)

Reply:
That's what I meant by the "we would need to know how many robots there are" part, as in like you need to know how many roles to make.

Reply:
Sure, the idea was just that we request more roles than will be fillable and you only fill some of them. But sure, the role request function can accept a world state.

Reply:
Yeah, but like how many roles do you actually request? 5? 10? 15? That's dependent on how many robots you actually have right?